### PR TITLE
CLOUDP-62037: Add Labels when Creating/Updating cluster with mongocli Atlas

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ [CLOUDP-62037](https://jira.mongodb.org/browse/CLOUDP-62037)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] I have tested my code using cloud-dev
## Further comments

**Create Cluster:**
```
./bin/mongocli atlas clusters create myCluster24 --provider AWS --region US_EAST_1 --members 3 --tier M10 --mdbVersion 4.2 --diskSizeGB 10
{
	"autoScaling": {
		"diskGBEnabled": true
	},
	"backupEnabled": false,
	"biConnector": {
		"enabled": false,
		"readPreference": "secondary"
	},
	"clusterType": "REPLICASET",
	"diskSizeGB": 10,
	"encryptionAtRestProvider": "NONE",
	"labels": [
		{
			"key": "Infrastructure Tool",
			"value": "mongoCLI"
		}
	],
	"id": "5ebd53141f8fbb684767ebaf",
	"groupId": "5e4e593f70dfbf1010295836",
	"mongoDBVersion": "4.2.6",
	"mongoDBMajorVersion": "4.2",
	"mongoURIUpdated": "2020-05-14T14:17:56Z",
	"name": "myCluster24",
	"numShards": 1,
	"paused": false,
	"pitEnabled": false,
	"providerBackupEnabled": false,
	"providerSettings": {
		"diskIOPS": 100,
		"encryptEBSVolume": true,
		"instanceSizeName": "M10",
		"providerName": "AWS",
		"regionName": "US_EAST_1",
		"volumeType": "STANDARD"
	},
	"replicationFactor": 3,
	"replicationSpec": {
		"US_EAST_1": {
			"analyticsNodes": 0,
			"electableNodes": 3,
			"priority": 7,
			"readOnlyNodes": 0
		}
	},
	"replicationSpecs": [
		{
			"id": "5ebd53141f8fbb684767ebaa",
			"numShards": 1,
			"zoneName": "Zone 1",
			"regionsConfig": {
				"US_EAST_1": {
					"analyticsNodes": 0,
					"electableNodes": 3,
					"priority": 7,
					"readOnlyNodes": 0
				}
			}
		}
	],
	"stateName": "CREATING",
	"connectionStrings": {}
}
```

**Update Cluster:**
```
./bin/mongocli atlas clusters update myCluster01 --diskSizeGB 20
{
	"autoScaling": {
		"diskGBEnabled": true
	},
	"backupEnabled": false,
	"biConnector": {
		"enabled": false,
		"readPreference": "secondary"
	},
	"clusterType": "REPLICASET",
	"diskSizeGB": 20,
	"encryptionAtRestProvider": "NONE",
	"labels": [
		{
			"key": "KeyUpdated",
			"value": "ValueUpdated"
		},
		{
			"key": "NewKey1",
			"value": "newValue1"
		},
		{
			"key": "Infrastructure Tool",
			"value": "mongoCLI"
		}
	],
	"id": "5ebd3d99669f2621a51e830b",
	"groupId": "5e4e593f70dfbf1010295836",
	"mongoDBVersion": "4.2.6",
	"mongoDBMajorVersion": "4.2",
	"mongoURI": "mongodb://mycluster01-shard-00-00.ajlj3.mongodb-dev.net:27017,mycluster01-shard-00-01.ajlj3.mongodb-dev.net:27017,mycluster01-shard-00-02.ajlj3.mongodb-dev.net:27017",
	"mongoURIUpdated": "2020-05-14T12:55:26Z",
	"mongoURIWithOptions": "mongodb://mycluster01-shard-00-00.ajlj3.mongodb-dev.net:27017,mycluster01-shard-00-01.ajlj3.mongodb-dev.net:27017,mycluster01-shard-00-02.ajlj3.mongodb-dev.net:27017/?ssl=true\u0026authSource=admin\u0026replicaSet=atlas-2ejb43-shard-0",
	"name": "myCluster01",
	"numShards": 1,
	"paused": false,
	"pitEnabled": false,
	"providerBackupEnabled": false,
	"providerSettings": {
		"diskIOPS": 100,
		"encryptEBSVolume": true,
		"instanceSizeName": "M10",
		"providerName": "AWS",
		"regionName": "US_EAST_1",
		"volumeType": "STANDARD"
	},
	"replicationFactor": 3,
	"replicationSpec": {
		"US_EAST_1": {
			"analyticsNodes": 0,
			"electableNodes": 3,
			"priority": 7,
			"readOnlyNodes": 0
		}
	},
	"replicationSpecs": [
		{
			"id": "5ebd3d99669f2621a51e8306",
			"numShards": 1,
			"zoneName": "Zone 1",
			"regionsConfig": {
				"US_EAST_1": {
					"analyticsNodes": 0,
					"electableNodes": 3,
					"priority": 7,
					"readOnlyNodes": 0
				}
			}
		}
	],
	"srvAddress": "mongodb+srv://mycluster01.ajlj3.mongodb-dev.net",
	"stateName": "UPDATING",
	"connectionStrings": {
		"standard": "mongodb://mycluster01-shard-00-00.ajlj3.mongodb-dev.net:27017,mycluster01-shard-00-01.ajlj3.mongodb-dev.net:27017,mycluster01-shard-00-02.ajlj3.mongodb-dev.net:27017/?ssl=true\u0026authSource=admin\u0026replicaSet=atlas-2ejb43-shard-0",
		"standardSrv": "mongodb+srv://mycluster01.ajlj3.mongodb-dev.net"
	}
}
```

### CODE REVIEW

File: internal/cli/atlas_clusters_update.go [hunk 1]
```diff
@@ -98,6 +98,25 @@ func (opts *atlasClustersUpdateOpts) patchOpts(out *atlas.Cluster) {
...
+func (opts *atlasClustersUpdateOpts) updateLabels(out *atlas.Cluster) {
+	found := false
+	for _, v := range out.Labels {
+		if v.Key == config.LabelKey && v.Value == config.LabelValue {
+			found = true
+			break
+		}
+	}
+
+	if !found {
```
Comment (andreaangiolillo) [inline internal/cli/atlas_clusters_update.go:114]: @gssbzn I know you said always but I do not see the use to put the same label if it is already there. Let me know if I have to change this.

File: internal/config/const.go [hunk 2]
```diff
@@ -29,4 +29,6 @@ const (
 	privateAPIKey       = "private_api_key"
 	opsManagerURL       = "ops_manager_url"
 	baseURL             = "base_url"
+	LabelKey            = "Infrastructure Tool"
```
Comment (gssbzn) [inline internal/config/const.go:32]: [nit] this is not part of the cli configuration so let's keep it in the cli package

File: internal/cli/atlas_clusters_update.go (hunk 1 above)
Comment (gssbzn) [reply internal/cli/atlas_clusters_update.go:114]: this is correct, that's why I mentioned as a gotcha in my original comment
One small nit is that we should follow this same approach for create as well, someone could be creating a cluster with a file and defining their own labels so we need to do this kind of check/append for both update and create

File: internal/cli/atlas_clusters_create.go [hunk 3]
```diff
@@ -85,17 +85,29 @@ func (opts *atlasClustersCreateOpts) newCluster() (*atlas.Cluster, error) {
...
-			Value: config.LabelValue,
-		},
-	}
+	opts.updateLabels(cluster)
 
 	cluster.GroupID = opts.ProjectID()
 	return cluster, nil
 }
 
+func (opts *atlasClustersCreateOpts) updateLabels(out *atlas.Cluster) {
```
Comment (gssbzn) [inline internal/cli/atlas_clusters_create.go:94]: [nit] this method doesn't use `opts` in its body, so you could make it a generic function like 
```suggestion
func  updateLabels(out *atlas.Cluster) {
```
This also let's you reuse the same function in both create an update

**Pull request merged**